crates/dwallet-mpc-centralized-party/src/lib.rs

```diff
@@ -107,16 +113,34 @@ pub fn create_dkg_output(
         bcs::from_bytes(&decentralized_first_round_public_output)?;
     match decentralized_first_round_public_output {
         VersionedDwalletDKGFirstRoundPublicOutput::V1(decentralized_first_round_public_output) => {
-            let (decentralized_first_round_public_output, _): <<AsyncProtocol as Protocol>::EncryptionOfSecretKeyShareRoundParty as Party>::PublicOutput =
+            let [first_part, second_part]: <DWalletDKGFirstParty as Party>::PublicOutput =
                 bcs::from_bytes(&decentralized_first_round_public_output)
                     .context("failed to deserialize decentralized first round DKG output")?;
-
+            // This is a temporary hack to keep working with the existing 2-round dWallet DKG mechanism.
+            // TODO (#1470): Use one network round in the dWallet DKG flow.
+            let protocol_pp_with_decentralized_dkg_output = ProtocolPublicParameters::new::<
+                { group::secp256k1::SCALAR_LIMBS },
+                SECP256K1_FUNDAMENTAL_DISCRIMINANT_LIMBS,
+                SECP256K1_NON_FUNDAMENTAL_DISCRIMINANT_LIMBS,
+                group::secp256k1::GroupElement,
+            >(
+                first_part.1,
+                second_part.1,
+                first_part.0,
+                second_part.0,
+                public_parameters
+                    .encryption_scheme_public_parameters
+                    .clone(),
+            );
             let session_identifier = CommitmentSizedNumber::from_le_slice(&session_identifier);
-
             let round_result = DKGCentralizedParty::advance(
-                decentralized_first_round_public_output,
+                (),
                 &(),
-                &(public_parameters, session_identifier).into(),
+                &(
+                    protocol_pp_with_decentralized_dkg_output,
+                    session_identifier,
+                )
+                    .into(),
                 &mut OsCsRng,
             )
             .context("advance() failed on the DKGCentralizedParty")?;
```

Comment:
there's another ugly part you are missing for both the user and network: when you get back the public output, it is already in the `VersionedOutput::UniversalPublicDKGOutput`. But this was a hack, it isn't really the case, and instead we should have it be as the old V1 `output` field inside it.

Later, when V1 (network version) wallets are being deserialized, they should be deserialized as `Output` and not `VersionedOutput`, and then you use `.into()` to transform them into a `VersionedOutput::TargetedPublicDKGOutput`

Reply:
good it was fixed